python/python/lance/vector.py

```diff
@@ -141,12 +141,12 @@ def train_ivf_centroids(
         )
 
     total_size = dataset.count_rows()
-    sample_size = min(k * sample_rate, total_size)
+    sample_size = k * sample_rate
 
-    if total_size < k * sample_size:
+    if total_size <= sample_size:
         samples = dataset.to_table(columns=[column])[column].combine_chunks()
     else:
-        samples = dataset.sample(k * sample_rate)[column]
+        samples = dataset.sample(sample_size, columns=[column])[column].combine_chunks()
```

Comment:
do we still need `[columns]` here?

Reply:
Yes.  `dataset.sample` returns a table, even if we are only sampling one column.

```
dataset.sample(sample_size, columns=[column]) -> pa.Table
...[column] -> pa.ChunkedArray
....combine_chunks() -> pa.Array
```